src/server/ApiRoutes/SubjectHeadings.js

```diff
@@ -0,0 +1,96 @@
+import axios from 'axios';
+
+import Bib from './Bib';
+import logger from '../../../logger';
+import appConfig from '../../../appConfig';
+
+const convertShepBibsToDiscoveryBibs = (response) => {
+  return Promise.all(
+    response.data.bibs.map((bib) => {
+      // Determine relevant id prefix
+      const institutionCode = {
+        'recap-pul': 'pb',
+        'recap-cul': 'cb',
+      }[bib.institution] || 'b';
+
+      const prefixedIdentifier = [institutionCode, bib.bnumber].join('');
+
+      return Bib.nyplApiClientCall(prefixedIdentifier)
```

Comment:
Do you think it would be worth building an endpoint that would accept a list of bib ids and return the same response that is generated here from multiple API calls?

Reply:
Possibly. I haven't had a chance to try the query and see if it's likely to be faster. I'll try to do that shortly.

Reply:
So, while it would be possible to add support for filtering via multiple "bibids" to the DiscoveryAPI (as a new supported filter parameter on the existing "resources" endpoint), I'm not sure it would make a big difference on the worst delays. We would absolutely see a small savings - on the order of half a second in some cases. But it looks to me like the pages where bibs load in slowest are those SHs with lots of titles, and the slowness is mostly occurring in the SHEP api. Here's timing logging on "Knowledge, Theory of":
```
Handling SHEP API Call: /subject_headings/8c754fce-2397-4a05-a5b2-017e4ad4fb1c/context
Handling SHEP API Call: /subject_headings/8c754fce-2397-4a05-a5b2-017e4ad4fb1c/bibs
Handling SHEP API Call: /subject_headings/8c754fce-2397-4a05-a5b2-017e4ad4fb1c/related
Timing: Time to perform shepApiCall (/subject_headings/8c754fce-2397-4a05-a5b2-017e4ad4fb1c/related):  74
Timing: Time to perform shepApiCall (/subject_headings/8c754fce-2397-4a05-a5b2-017e4ad4fb1c/context):  380
Timing: Time to perform shepApiCall (/subject_headings/8c754fce-2397-4a05-a5b2-017e4ad4fb1c/bibs):  8702
Timing: Time to convertShepBibsToDiscoveryBibs:  203
```

The "bibs" call on the SHEP API takes 8.7s. The time taken to hit the DiscoveryAPI to convert those bibs to fully formed DiscoveryAPI resources via multiple calls to the DiscoveryAPI (i.e. `convertShepBibsToDiscoveryBibs`) is 203ms. The addition of a new "bibids" filter param might reduce that by 90%, but SH pages with lots of titles would still load titles slowly.

So although I've come around to agreeing a bibids filter param would be a useful feature, I don't think it's a high priority for this work right now. I think we need to find ways to optimize the bibs endpoint on the SHEP API